src/pointcloud_to_laserscan_node.cpp

```diff
@@ -225,9 +243,80 @@ void PointCloudToLaserScanNode::cloudCallback(
     int index = (angle - scan_msg->angle_min) / scan_msg->angle_increment;
     if (range < scan_msg->ranges[index]) {
       scan_msg->ranges[index] = range;
+      v_pointcloud_index.at(index) = pointcloud_index;
     }
   }
-  pub_->publish(std::move(scan_msg));
+
+  laserscan_pub_->publish(std::move(scan_msg));
+
+  pcl::PointCloud<pcl::PointXYZ> pcl_pointcloud;
+  sensor_msgs::PointCloud2ConstIterator<float> iter_x(*cloud_msg, "x"), iter_y(*cloud_msg, "y"),
+    iter_z(*cloud_msg, "z");
+  for (size_t i = 0; i < v_pointcloud_index.size(); ++i) {
+    if (v_pointcloud_index.at(i) != no_data) {
+      pcl::PointXYZ point;
+      point.x = *(iter_x + v_pointcloud_index.at(i));
+      point.y = *(iter_y + v_pointcloud_index.at(i));
+      point.z = *(iter_z + v_pointcloud_index.at(i));
+      pcl_pointcloud.push_back(point);
+    }
+  }
+  pcl::toROSMsg(pcl_pointcloud, pointcloud_output);
+  auto pointcloud_output_ptr
+    = std::make_unique<sensor_msgs::msg::PointCloud2>(pointcloud_output);
```

Comment:
[nits] `sensor_msgs::msg::PointCloud2` can be abbreviated? Then you can write in one line.

Reply:
Fixed at https://github.com/tier4/pointcloud_to_laserscan/pull/1/commits/3371ea3e610f4e4dfec523fdada45c11ada3d888

Reply:
Oh, `using` is nice!